### PR TITLE
add osa origin conformance e2e test via base images

### PR DESCRIPTION
**Pull request by mjudeikis**
Add e2e origin test to OSA cluster

### CODE REVIEW

**Comment by mjudeikis** [comment]
@kargakis @jim-minter ready for first review

File: ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+      emptyDir: {}
+    - name: cluster-secrets-azure
+      secret:
+        secretName: e2e-azure-secret
+
+    containers:
+
+    # Once admin.kubeconfig exists, executes shared tests
+    - name: test
+      image: registry.svc.ci.openshift.org/azure/origin-test:v3.10
```
Comment (kargakis) [inline ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: Why not use the upstream openshift/origin-tests image?

File: ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        echo "Found shared kubeconfig"
+
+        # test specific code
+        # don't let clients impact the global kubeconfig
+        cp -r /tmp/shared/_data /tmp/openshift/
+        export KUBECONFIG=/tmp/openshift/_data/_out/admin.kubeconfig
+
+        PATH=/usr/libexec/origin:$PATH
+        # create the environment, but exit before we run tests
+        # TODO: RUN TESTS HERE
```
Comment (kargakis) [inline ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: remove

File: ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        oc get pods --all-namespaces
```
Comment (kargakis) [inline ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: remove

File: ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+        cd /go/src/github.com/openshift/origin/
+        export LOG_DIR=/tmp/openshift/logs
+        export VOLUME_DIR=/tmp/openshift/volume
+        export ARTIFACT_DIR=/tmp/openshift/ARTIFACT
+        export FAKE_HOME_DIR=/tmp/openshift/home
+
+        TEST_ONLY=1 test/extended/core.sh > /tmp/artifacts/origin_tests
+        
+        exit 0
```
Comment (kargakis) [inline ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: not needed?

File: ci-operator/templates/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 5]
```diff
@@ -112,6 +112,7 @@ objects:
 
         # create the environment, but exit before we run tests
         oc get pods --all-namespaces 
+        exit 0
```
Comment (kargakis) [inline ci-operator/templates/openshift-azure/cluster-launch-e2e-azure.yaml]: why is this needed?

File: ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+  metadata:
+    name: ${JOB_NAME_SAFE}-image-puller
+    namespace: ${NAMESPACE}
+  roleRef:
+    name: system:image-puller
+  subjects:
+  - kind: SystemGroup
+    name: system:unauthenticated
+
+# The e2e pod spins up a cluster, runs e2e tests, and then cleans up the cluster.
```
Comment (kargakis) [inline ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: comment needs update

File: ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml [hunk 7]
```diff
@@ -0,0 +1,236 @@
+# This template is being used for testing OpenShift on Azure. 
+# At the moment it cannot be reused to launch cluster for Origin tests
```
Comment (kargakis) [inline ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: is this valid now?

Comment (mjudeikis) [reply ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: I think so. But in different form :) will move

File: ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml (hunk 1 above)
Comment (mjudeikis) [reply ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: need to check it if it does what we need to do. Never used it.

Comment (mjudeikis) [reply ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: The image does not container uncompiled tests and code :/ does not work for us. it does not even has go in it.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-presubmits.yaml [hunk 8]
```diff
@@ -178,3 +178,59 @@ presubmits:
...
+        - --template=/usr/local/e2e-azure-conformance
+        - --target=[output:stable:sync]
+        - --target=e2e-azure-conformance
+        command:
+        - ci-operator
+        env:
+        - name: CONFIG_SPEC
+          valueFrom:
+            configMapKeyRef:
+              key: master.json
```
Comment (petr-muller) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-presubmits.yaml]: This needs to be changed to `master.yaml`.

We have mass-renamed all ci-operator configuration files from `.json` to `.yaml` in https://github.com/openshift/release/pull/1336, because YAML is what ci-operator consumes now. JSON is a YAML subset so that the actual file content can stay like it is (we'll eventually mass-reformat them to the block style).

File: ci-operator/templates/openshift-azure/README.md [hunk 9]
```diff
@@ -0,0 +1,2 @@
+# These templates is being used for testing OpenShift on Azure.
```
Comment (kargakis) [inline ci-operator/templates/openshift-azure/README.md]: s/is/are/

File: ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml (hunk 1 above)
Comment (kargakis) [reply ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: It doesn't need to have go, does it? It should contain a binary with all the Origin conformance tests.

Comment (mjudeikis) [reply ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: hmmm. found them `./ginkgo extended.test`  need to check if it contains all what we need. would be good to swap

File: ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml [hunk 10]
```diff
@@ -0,0 +1,221 @@
+kind: Template
```
Comment (smarterclayton) [inline ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: Why do you have cluster-launch-e2e-azure and cluster-launch-e2e-azure-conformance?

Comment (kargakis) [reply ci-operator/templates/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: The first is meant to run openshift on azure specific e2e tests, the second is meant to run the origin conformance suite on a cluster deployed on azure.

**Comment by kargakis** [comment]
Let's see what falls out and fix it

/lgtm
